content/en/docs/12/05/_index.en.md

```diff
@@ -0,0 +1,248 @@
+
+In the test output you will some issues regarding the sample plugins, feel free to either remove these plugins or ignore them.
+
+Does the `schroedingers_cat` plugin pass the sanity tests?
+
+Try to fix all the tests.
+
+{{% details title="Solution Task 2" %}}
+
+You can execute a specific sanity test by running:
+
+```bash
+tox -e sanity-py3.12-2.18 --ansible --conf tox-ansible.ini
+```
+
+Those plugins can be removed:
+
+```bash
+rm plugins/test/sample_test.py
+rm plugins/action/sample_action.py
+rm plugins/action/sample_*.py
+```
+
+{{% /details %}}
+
+### Task 3 - Unit Testing our collection
+
+Unit tests can be run similarly to sanity tests but with the `unit` environment:
+
+```bash
+$ tox -e unit-py3.12-2.18 --ansible --conf tox-ansible.ini
+....
+
+2 workers [1 item]      
+scheduling tests via LoadScheduling
+
+tests/unit/test_basic.py::test_basic 
+[gw0] [100%] PASSED tests/unit/test_basic.py::test_basic 
+
+================================================ 1 passed in 0.40s ================================================
+  unit-py3.12-2.18: OK (74.18=setup[45.34]+cmd[0.00,0.13,27.87,0.02,0.82] seconds)
+  congratulations :) (74.27 seconds)
+
+```
+
+As you can see from the output of this command there are already some basic tests present in the collection.
+We will ignore them for now and focus on writing our own tests against our collection to make sure our module works as expected.
+
+The Ansible Community Documentation has an extensive documentation page about unit testing modules that you can find [here](https://docs.ansible.com/ansible/latest/dev_guide/testing_units_modules.html).
+Keep this page in mind since it may come in handy as a reference for the following tasks.
+It may also be helpful if you are not familiar with unit testing and want to learn more on what unit tests are and why we need them.
+
+For now, we will start by creating a test file for the `schroedingers_cat` module in the `tests/unit/plugins/modules` directory and adding some test cases in it.
+In this lab we will use pytest style test cases.
+
+Can you implement the following unit test cases?
+
+1. The module should always return a `cat_state` on exit.
+2. The module should always return a cat state of `dead and alive` when the `force_box_open` argument is set to `False` (default).
+3. The module should always return a cat state of `alive` or `dead` when the `force_box_open` argument is set to `True`.
+
+
+{{% details title="Solution Task 3" %}}
+
+First create the unit test file:
+
+```bash
+cd training.labs
+mkdir -p tests/unit/plugins/modules
+touch tests/unit/plugins/{,modules/}__init__.py
+touch tests/unit/plugins/modules/test_schroedingers_cat.py
+```
+
+To write isolated unit tests we will need some of the utilities introduced in the Ansible Community Documentation.
+First we need to address the question how can i control with what arguments my module will be called in the unit test.
```

Comment:
First, we need to address the question of how we can control with which arguments my module will be called with in the unit test.